### PR TITLE
[magpietts] adhere CapWords convention for model class names.

### PR DESCRIPTION
**Pull request by XuesongYang**
> [!IMPORTANT]  
> The `Update branch` button must only be pressed in very rare occassions.
> An outdated branch is never blocking the merge of a PR.
> Please reach out to the automation team before pressing that button.

# What does this PR do ?
1. removed duplicated freezing operations for models that subclassed from `LightningModule` since it has already defined function as in https://lightning.ai/docs/pytorch/stable/_modules/lightning/pytorch/core/module.html#LightningModule.freeze, which combines false grad and eval together.
```python  
    def freeze(self) -> None:
        r"""Freeze all params for inference.

        Example::

            model = MyLightningModule(...)
            model.freeze()

        """
        for param in self.parameters():
            param.requires_grad = False

        self.eval()
```
2. adhered to the CapWords convention for class names before it is too late to change naming after we release models.
3. removed `def _setup_train_dataloader()`, and unified to `def setup_train_data()` for simplicity.

**Collection**: [Note which collection this PR will affect]

# Changelog

- Add specific line by line info of high level changes in this PR.

# Usage

- You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"

**Pre checks**:

- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:

- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed.
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information

- Related to # (issue)

### CODE REVIEW

File: nemo/collections/tts/models/magpietts.py [hunk 1]
```diff
@@ -23,22 +23,25 @@
 import soundfile as sf
 import torch
 from hydra.utils import instantiate
-from omegaconf import DictConfig, open_dict
 from lightning.pytorch import Trainer
-from pytorch_lightning.loggers import TensorBoardLogger
+from lightning.pytorch.loggers import TensorBoardLogge
```
Comment (blisc) [inline nemo/collections/tts/models/magpietts.py]: Can you undo this typo?

**Comment by blisc** [comment]
Why CapWords? Is there a doc stating that is the preferred style?

**Comment by XuesongYang** [comment]
> Why CapWords? Is there a doc stating that is the preferred style?

I remembered we regret some model classes not being the CapConvention before. So better to use convention as https://google.github.io/styleguide/pyguide.html#s3.16.2-naming-conventions.

If you search the pattern like `class .*_.*:`, it's very rare to see non-capwords class name in NeMo codebase.

File: nemo/collections/tts/models/magpietts.py [hunk 2]
```diff
@@ -206,17 +206,14 @@ def __init__(self, cfg: DictConfig, trainer: 'Trainer' = None):
         codec_model = AudioCodecModel.restore_from(cfg.get('codecmodel_path'), strict=False)
         # del codec discriminator to free memory
         del codec_model.discriminator
-        codec_model.eval()
-        self.freeze_model(codec_model)
         self._codec_model = codec_model
+        self._codec_model.freeze()
```
Comment (blisc) [inline nemo/collections/tts/models/magpietts.py]:
```suggestion
        self._codec_model.freeze()  #Lightning does requires_grad = False and self.eval()
```

File: nemo/collections/tts/models/magpietts.py [hunk 3]
```diff
@@ -206,17 +206,14 @@ def __init__(self, cfg: DictConfig, trainer: 'Trainer' = None):
...
 
         if self.model_type == 'single_encoder_sv_tts':
-            speaker_verification_model = nemo_asr.models.EncDecSpeakerLabelModel.from_pretrained(
+            self._speaker_verification_model = nemo_asr.models.EncDecSpeakerLabelModel.from_pretrained(
                 model_name='titanet_large'
             )
-            speaker_verification_model.eval()
-            self.freeze_model(speaker_verification_model)
-            self._speaker_verification_model = speaker_verification_model
+            self._speaker_verification_model.freeze()
```
Comment (blisc) [inline nemo/collections/tts/models/magpietts.py]:
```suggestion
            self._speaker_verification_model.freeze()  #Lightning does requires_grad = False and self.eval()
```
